p2p/simulator/main.go

```diff
@@ -0,0 +1,186 @@
+}
+
+var messages = []string{
+	"hello friend!\n",
+	"i am a node\n",
+	"do you want to be friends?\n",
+	"ok\n",
+	"noot\n",
+	"i am a penguin stuck in a computer\n",
+	"pls feed me code\n",
+}
+
+func StartIpfsNode() (*ipfs.IpfsNode, error) {
+	id := testIdentity
+
+	c := &config.Config{
+		Identity: id,
+		Addresses: config.Addresses{
+			Swarm: []string{"/ip4/0.0.0.0/tcp/4001"},
+			API:   []string{"/ip4/127.0.0.1/tcp/8000"},
+		},
+	}
+
+	r := &repo.Mock{
+		C: *c,
+		D: syncds.MutexWrap(datastore.NewMapDatastore()),
+	}
+
+	cfg := &ipfs.BuildCfg{
+		Online:    true,
+		Host:      ipfs.DefaultHostOption,
+		Repo:      r,
+		Permanent: true,
+	}
+
+	node, err := ipfs.NewNode(context.Background(), cfg)
+	return node, err
+}
+
+func NewSimulator(num int) (sim *Simulator, err error) {
+	sim = new(Simulator)
+	sim.nodes = make([]*p2p.Service, num)
+
+	// start local ipfs daemon
+	ipfsNode, err := StartIpfsNode()
+	if err != nil {
+		log.Fatalf("Could not start IPFS node: %s", err)
+	}
+
+	sim.ipfsNode = ipfsNode
+
+	ipfsAddr := fmt.Sprintf("/ip4/127.0.0.1/tcp/4001/ipfs/%s", ipfsNode.Identity.String())
+	log.Println("ipfsAddr:", ipfsAddr)
+
+	// create all nodes, increment port by 1 each time
+	for i := 0; i < num; i++ {
+		// configure p2p service
+		conf := &p2p.ServiceConfig{
+			BootstrapNodes: []string{
+				ipfsAddr,
+			},
+			Port: 5000 + i,
+		}
+		sim.nodes[i] = new(p2p.Service)
+		sim.nodes[i], err = p2p.NewService(conf)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return sim, nil
+}
+ 
+func sendRandomMessage(s *p2p.Service, peerid peer.ID) error {
+	// open new stream with each peer
+	ps, err := s.DHT().FindPeer(s.Ctx(), peerid)
+	if err != nil {
+		return fmt.Errorf("could not find peer: %s", err)
+	}
+
+	r := getRandomInt(len(messages))
+	msg := messages[r]
+
+	err = s.Send(ps, []byte(msg))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getRandomInt(m int) int {
+	b := make([]byte, 1)
+	_, err := rand.Read(b)	
+	if err != nil {
+		return 0
+	}
+	r := int(b[0]) % m
+	return r
+}
+
+func main() {
+	golog.SetAllLoggers(gologging.INFO) // Change to DEBUG for extra info
+
+	if len(os.Args) < 2 {
+		fmt.Println("please specify number of nodes to start in simulation: ./p2p/simulator/main.go [num]")
```

Comment:
avoid fmt.Println or Print use logger for all instances

Reply:
ok

Reply:
fixed